src/values/ECMAScriptSourceFunctionValue.js

```diff
@@ -23,6 +23,8 @@ export default class ECMAScriptSourceFunctionValue extends ECMAScriptFunctionVal
   $Strict: boolean;
   $FormalParameters: Array<BabelNodeLVal>;
   $ECMAScriptCode: BabelNodeBlockStatement;
+  $HasComputedName: boolean;
```

Comment:
I am not convinced that `$HasComputedName` and `$HasEmptyConstructor` are set along all paths where an instance of this class is created. Thus, the type should probably be `?: boolean`.